catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/DeleteOperations.java

```diff
@@ -267,6 +280,50 @@ private DeleteResponse performLocalDelete(DeleteRequest deleteRequest,
         return deleteResponse;
     }
 
+    //
+    // Private helper methods
+    //
+
+    private DeleteRequest rewriteRequestToAvoidHistoryConflicts(DeleteRequest deleteRequest,
+            SourceResponse response) {
+        if (Metacard.ID.equals(deleteRequest.getAttributeName())) {
+            return deleteRequest;
+        }
+
+        List<Serializable> updatedList = response.getResults()
+                .stream()
+                .map(Result::getMetacard)
+                .map(Metacard::getId)
+                .map(Serializable.class::cast)
+                .collect(Collectors.toList());
+
+        return new DeleteRequestImpl(updatedList,
+                Metacard.ID,
+                deleteRequest.getProperties(),
+                deleteRequest.getStoreIds());
+    }
+
+    private boolean notFoundValidMetacards(DeleteRequest deleteRequest, SourceResponse response) {
+        Set<String> originalKeys = deleteRequest.getAttributeValues()
+                .stream()
+                .map(Object::toString)
+                .collect(Collectors.toSet());
+        Set<String> responseKeys = response.getResults()
+                .stream()
+                .map(Result::getMetacard)
+                .map(m -> m.getAttribute(deleteRequest.getAttributeName()))
+                .filter(Objects::nonNull)
+                .map(Attribute::getValue)
```

Comment:
Only dealing with single-valued attributes? Am I worrying about something that's irrelevant to this change?